autopilot/contractor.go

```diff
@@ -1062,20 +1091,24 @@ func (c *contractor) refreshContract(ctx context.Context, w Worker, ci contractI
 
 	// do not refresh if the contract's updated collateral will fall below the threshold anyway
 	_, hostMissedPayout, _, _ := rhpv2.CalculateHostPayouts(rev.FileContract, newCollateral, settings, contract.EndHeight())
-	if isBelowCollateralThreshold(newCollateral, hostMissedPayout) {
-		err := fmt.Errorf("refresh failed, refreshed contract collateral (%v) is below threshold", hostMissedPayout)
+	var newRemainingCollateral types.Currency
+	if hostMissedPayout.Cmp(settings.ContractPrice) > 0 {
+		newRemainingCollateral = hostMissedPayout.Sub(settings.ContractPrice)
+	}
+	if isBelowCollateralThreshold(newCollateral, newRemainingCollateral) {
+		err := fmt.Errorf("discarding contract after a failed refresh, refreshed contract collateral (%v) is below threshold", hostMissedPayout)
 		c.logger.Errorw(err.Error(), "hk", hk, "fcid", fcid, "newCollateral", newCollateral.String(), "hostMissedPayout", hostMissedPayout.String(), "maxCollateral", settings.MaxCollateral)
-		return api.ContractMetadata{}, true, err
+		return api.ContractMetadata{}, true, true, err
```

Comment:
How about a custom error that we check for instead of a boolean? e.g. `ErrBelowCollateralThreshold`?